### PR TITLE
Refactor tests to create a fast local dev loop

### PR DESCRIPTION
**Pull request by pbruski**

### CODE REVIEW

File: src/test/kotlin/com/atlassian/performance/tools/jiraactions/api/scenario/JiraCoreScenarioDevloop.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import org.openqa.selenium.chrome.ChromeOptions
+import org.openqa.selenium.logging.LogType
+import org.openqa.selenium.logging.LoggingPreferences
+import org.openqa.selenium.remote.CapabilityType
+import org.openqa.selenium.remote.RemoteWebDriver
+import java.net.URI
+import java.util.logging.Level
+
+/**
+ * This class is meant to be used in a local devloop, where JIra is already running.and Chrome is locally available.
```
Comment (dagguh) [inline src/test/kotlin/com/atlassian/performance/tools/jiraactions/api/scenario/JiraCoreScenarioDevloop.kt:17]: Oooh, I get it. We can have tests that work locally and not on CI.
Not having to provision Jira every time is gonna be faster for sure.
OTOH the dev still has to provision manually Jira at least once, so the first loop is still slow (and tribal). We need Jira to start quickly. What's the duration of `JiraCoreFormula.provision()`?

Comment (pbruski) [reply src/test/kotlin/com/atlassian/performance/tools/jiraactions/api/scenario/JiraCoreScenarioDevloop.kt:17]: I have data for the full run - it's 12 minutes vs 1m30s . Obviously, if you need to test just one action, it becomes 12minutes vs 10-20 seconds...

Comment (dagguh) [reply src/test/kotlin/com/atlassian/performance/tools/jiraactions/api/scenario/JiraCoreScenarioDevloop.kt:17]: That's convincing!

Comment (dagguh) [reply src/test/kotlin/com/atlassian/performance/tools/jiraactions/api/scenario/JiraCoreScenarioDevloop.kt:17]: Do you need this for ShipIt? I'd like to review some small stuff here, but I'm participating in the ShipIt right now, so I'm deferring PRs by default

Comment (pbruski) [reply src/test/kotlin/com/atlassian/performance/tools/jiraactions/api/scenario/JiraCoreScenarioDevloop.kt:17]: No, keep shipping!

Comment (mzyromski-atlassian) [reply src/test/kotlin/com/atlassian/performance/tools/jiraactions/api/scenario/JiraCoreScenarioDevloop.kt:17]: As this test is intended for manual execution only it won't get executed on CI which makes it prone to degradation in the future. That said, the gain for a dev is so big that it indeed might make sense to have it.

Comment (mzyromski-atlassian) [inline src/test/kotlin/com/atlassian/performance/tools/jiraactions/api/scenario/JiraCoreScenarioDevloop.kt:17]: nit: typos ("JIra", dot before "and")

File: src/test/kotlin/com/atlassian/performance/tools/jiraactions/api/scenario/JiraCoreScenarioIT.kt [hunk 2]
```diff
@@ -32,111 +15,18 @@ class JiraCoreScenarioIT {
...
-                return user
-            }
-
-            override fun remember(memories: Collection<User>) {
-                throw Exception("not implemented")
-            }
-        }
-        var firstBackupElement = true
-        var secondBackupElement = true
+        val jiraVersion = System.getenv("JIRA_SOFTWARE_VERSION") ?: "8.0.0"
```
Comment (mzyromski-atlassian) [inline src/test/kotlin/com/atlassian/performance/tools/jiraactions/api/scenario/JiraCoreScenarioIT.kt:18]: How about accessing the value from `AbstractJiraCoreScenario`? I'm afraid that having it here and there (where it's actually only used to display a log message) will lead to people making mistakes in the future when e.g. changing a default value.

Comment (pbruski) [reply src/test/kotlin/com/atlassian/performance/tools/jiraactions/api/scenario/JiraCoreScenarioIT.kt:18]: Sure!

**Pull request merged**